test/units/utils/collection_loader/test_collection_loader.py

```diff
@@ -37,7 +37,9 @@ def test_find_module_py3():
     # setuptools may fall back to find_module on Python 3 if find_spec returns None
     # see https://github.com/pypa/setuptools/pull/2918
     assert path_hook_finder.find_spec('missing') is None
-    assert path_hook_finder.find_module('missing') is None
+    if sys.version_info < (3, 12):
+        # find_module has been removed in Python 3.12
```

Comment:
Since this generates a deprecation warning on Python 3.11, we should suppress that here.